### PR TITLE
port to c++11

### PR DESCRIPTION
**Pull request by sandsmark**
removes the mandatory boost dependency, only thing left is boost::python, which ideally would be replaced with pybind11 (helped dlib quite a bit when it did).

### CODE REVIEW

**Comment by sandsmark** [comment]
https://github.com/davisking/dlib/issues/293 fwiw

**Comment by HannesSommer** [comment]
ok to test

File: tests/helpers.h [hunk 1]
```diff
@@ -129,55 +125,20 @@ typename NearestNeighbourSearch<T>::Matrix createQuery(const typename NearestNei
...
-#include <mach/mach.h>
-#else
-#include <time.h>
-#endif
-
-#ifdef _POSIX_TIMERS 
-namespace boost
+#include <chrono>
+using std::chrono::high_resolution_clock;
+using std::chrono::duration_cast;
```
Comment (HannesSommer) [inline tests/helpers.h]: I'd rather use explicit scoping or function level `using`  below.

**Comment by sandsmark** [comment]
Hi!

Yeah, I actually wanted to do what you suggested, but I don't have my laptop here, so feel free to do it. Otherwise I can do it when I get home. :-)

**Pull request merged**